Register beta prototype v1 (#247)
DFE-Digital/bat-design-history: app/posts/register-trainee-teachers/2020-09-30-beta-prototype.md

---
title: Beta prototype v1
description:
date: 2020-10-12
related:
  items:
  - text: Current beta prototype (Heroku)
    href: https://register-prototype.herokuapp.com/
  - text: Github repo for prototype
    href: https://github.com/DFE-Digital/register-trainee-teachers-prototype
  - text: Alpha prototype
    href: /register-trainee-teachers/alpha-prototype/
---

> ## Building a new prototype from scratch

Reviewer: Not sure you need this heading – perhaps it could be the summary for this post instead?

Author: 👍

> Starting in early September, we began work on a new prototype for the Register service. The [alpha prototype](/register-trainee-teachers/alpha-prototype/) was a static invision prototype, which while good for quick iteration, limited some of the usability testing we could do. This new one is built with the [GOV.UK Prototype Kit](https://govuk-prototype-kit.herokuapp.com/) and allows us to test more complex journeys with more realistic data.

Reviewer: ```suggestion
We began work on a new prototype for the Register service in early September 2020. The [alpha prototype](/register-trainee-teachers/alpha-prototype/) was a static InVision prototype, which while good for quick iteration, limited some of the usability testing we could do. This new one is built with the [GOV.UK Prototype Kit](https://govuk-prototype-kit.herokuapp.com/) and allows us to test complex journeys with realistic data.
```

## Journey changes

> Working in html and designing for progressive enhancement necessitated some changes from the alpha designs:
> * use the task list pattern rather than tabs for each form
> * break up forms across pages to support adding multiples (like multiple degrees)
> * implicitly require that each section be completed as a whole, and not allow individual fields to be left blank

Reviewer: ```suggestion
Working in HTML and designing for progressive enhancement necessitated some changes from the alpha designs:

* use the task list pattern rather than tabs for each form
* break up forms across pages to support adding multiples (like multiple degrees)
* implicitly require that each section be completed as a whole, and not allow individual fields to be left blank
```

> The new prototype borrows heavily from the Apply and Manage services. Register sits half way between them in sorts of data collected and actions users need to take - so it's been incredibly helpful to have them to take inspiration from.

Reviewer: ```suggestion
The new prototype borrows heavily from the Apply and Manage services. Register sits half way between them in term of data collected and actions users need to take, so it’s been incredibly helpful to have them to take inspiration from.
```

## Task list pattern

> This pattern allows us to have multiple sections that users can complete in any order. As Register trainee teachers supports more routes, we can vary the sections shown to be specific to the route.

Reviewer: ```suggestion
This pattern allows us to have multiple sections that users can complete in any order. As the Register service supports more routes, we can vary the sections shown to they are specific to each route.
```

{% from "figure/macro.njk" import appFigure with context %}

> Task list overview:
> {{ appFigure({
>   image: {
>     file: "2020.09.22_17_17_39_Overview - Register trainee teachers - GOV.UK.png"
>   }
> }) }}

Reviewer: Perhaps use captions for these figures?

```suggestion
{{ appFigure({
  image: {
    file: "2020.09.22_17_17_39_Overview - Register trainee teachers - GOV.UK.png"
  },
  caption: "Task list overview"
}) }}
```

}) }}

A partially completed task list:
{{ appFigure({
  image: {
    file: "2020.09.22_17_21_58_Overview - Register trainee teachers - GOV.UK.png"
  }
}) }}

## Screenshots of all pages

{% from "screenshots/macro.njk" import appScreenshots with context %}
{{ appScreenshots({
  hideContents: true,
  items: [{
      text: "2020.09.22_17_06_58_Register for teacher training   GOV.UK",
      img: { src: "2020.09.22_17_06_58_Register for teacher training - GOV.UK.png" }
    }, {
      text: "2020.09.22_17_12_55_  Register trainee teachers   GOV.UK",
      img: { src: "2020.09.22_17_12_55_- Register trainee teachers - GOV.UK.png" }
    }, {
      text: "2020.09.22_17_15_23_  Register trainee teachers   GOV.UK",
      img: { src: "2020.09.22_17_15_23_- Register trainee teachers - GOV.UK.png" }
    }, {
      text: "2020.09.22_17_15_45_Trainee teachers   Register trainee teachers   GOV.UK",
      img: { src: "2020.09.22_17_15_45_Trainee teachers - Register trainee teachers - GOV.UK.png" }
    }, {
      text: "2020.09.22_17_16_00_Add a trainee   Register trainee teachers   GOV.UK",
      img: { src: "2020.09.22_17_16_00_Add a trainee - Register trainee teachers - GOV.UK.png" }
    }, {

>       text: "2020.09.22_17_16_11_Other routes not supported   Register trainee teachers   GOV.UK",

Reviewer: Remove the date and suffixes.

```suggestion
      text: "Other routes not supported",
```
